### PR TITLE
[script.module.phate89@matrix] 1.2.1+matrix.1

### PR DESCRIPTION
**Pull request by phate89**
### Add-on details:

- **General**
  - Add-on name: phate89 helper module
  - Add-on ID: script.module.phate89
  - Version number: 1.2.1+matrix.1
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/phate89/script.module.phate89
  
Helper module for addons

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: script.module.phate89/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.module.phate89" name="phate89 helper module" version="1.2.1+matrix.1" provider-name="phate89">
+	<requires>
+		<import addon="xbmc.python" version="3.0.0"/>
+		<import addon="script.module.requests" version="2.9.1"/>
+		<import addon="script.module.beautifulsoup4" version="4.3.1" />
+		<import addon="script.module.simplejson" version="2.0.10"/>
```
Comment (enen92) [inline script.module.phate89/addon.xml:7]: drop

File: script.module.phate89/lib/phate89lib/kodiutils.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+import os
+import re
+import sys
+from kodi_six import xbmc, xbmcaddon, xbmcplugin, xbmcgui, utils  # pylint: disable=import-error
+try:
+    from urllib.parse import urlencode
+except ImportError:
+    from urllib import urlencode
+from . import staticutils
+if sys.version_info < (2, 7):
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/kodiutils.py:12]: no longer needed, use only json

File: script.module.phate89/lib/phate89lib/kodiutils.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+KODILANGUAGE = xbmc.getLocalizedString
+
+HANDLE = int(sys.argv[1])
+
+
+def executebuiltin(func, block=False):
+    xbmc.executebuiltin(func, block)
+
+
+def notify(msg):
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/kodiutils.py:36]: xbmcgui.Dialog().notification()

File: script.module.phate89/lib/phate89lib/kodiutils.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+def append_subtitle(sUrl, subtitlename, sync=False, provider=None):
+    #listitem = createListItem({'label': 'Italian', 'label2': subtitlename, 'thumbnailImage': 'it'})
+    if not provider:
+        tUrl = {'action': 'download', 'subid': sUrl}
+    elif provider == 'ItalianSubs':
+        p = re.search('subtitle_id=(?P<SUBID>[0-9]+)', sUrl, re.IGNORECASE)
+        if not p:
+            return False
+        tUrl = "plugin://service.subtitles.itasa/?action=download&subid={subid}".format(
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/kodiutils.py:148]: what is service.subtitles.itasa? This is not in matrix is it?

File: script.module.phate89/lib/phate89lib/kodiutils.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+            subid=p.group('SUBID'))
+    else:
+        tUrl = {'action': 'download', 'url': sUrl}
+    log("aggiungo il sottotitolo '" + subtitlename + "' alla lista", 3)
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/kodiutils.py:152]: please use english for log messages

File: script.module.phate89/lib/phate89lib/rutils.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            subs = [x for x in zf.namelist() if os.path.splitext(x)[1] in exts]
+            if index >= len(subs):
+                index = 0
+            binData = zf.read(subs[index])
+            zf.close()
+            outName = os.path.join(dataPath, subs[index])
+
+            if binData:
+                try:
+                    fp = open(outName, 'wb')
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/rutils.py:131]: use python's with statement:

```
with open(xxx) as fp:
    f.write()
```

File: script.module.phate89/lib/phate89lib/rutils.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+            try:
+                from kodi_six import xbmc
+            except ImportError:
+                self.log('rar extraction not supported', 1)
+                return False
+            TEMPFOLDER = os.path.join(dataPath, 'temp', '')
+            if os.path.isdir(TEMPFOLDER):
+                shutil.rmtree(TEMPFOLDER)
+            os.makedirs(TEMPFOLDER)
+            xbmc.executebuiltin('XBMC.Extract(%s,%s)' % (TEMPFILE, TEMPFOLDER), True)
```
Comment (enen92) [inline script.module.phate89/lib/phate89lib/rutils.py:151]: XBMC prefixes were dropped in matrix, use only Extract

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

**Comment by phate89** [comment]
Most of the changes was because I wanted to keep a unique code base but it's probably a little too difficult. I fixed all the suggestions (and branched a different version pre-matrix)
Thanks

**Review by enen92 — APPROVED** [review]

**Pull request merged**